File: app/src/main/java/com/cleanup/todoc/datasource/dao/TaskDao.java
package com.cleanup.todoc.datasource.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.cleanup.todoc.datasource.Entity.TaskEntity;

import java.util.List;

@Dao
public interface TaskDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(TaskEntity taskEntity);

    @Delete
    void delete(TaskEntity taskEntity);

    @Query("SELECT * FROM TaskEntity")
    LiveData<List<TaskEntity>> getTasks();

    @Query("SELECT * FROM TaskEntity ORDER BY projectId")
    LiveData<List<TaskEntity>> getTasksByProject();
}

File: app/src/main/java/com/cleanup/todoc/datasource/dao/TaskDao.kt
package com.cleanup.todoc.datasource.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.cleanup.todoc.datasource.entity.TaskEntity
import kotlinx.coroutines.flow.Flow

@Dao
interface TaskDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insert(taskEntity: TaskEntity)

    @Delete
    suspend fun delete(taskEntity: TaskEntity)

    @Query("SELECT * FROM TaskEntity")
    fun getTasks(): Flow<List<TaskEntity>>

    @Query("SELECT * FROM TaskEntity ORDER BY projectId")
    fun getTasksByProject(): Flow<List<TaskEntity>>
}
